helm/prometheus-meta-operator/templates/recording-rules/cortex.rules.yml

```diff
@@ -10,184 +10,180 @@ spec:
     # API usage: authentication fails via the "Bearer" auth method (staff)
-    - record: aggregation:giantswarm:api_auth_jwt_failed_attempts_total
-      expr: "sum(rate(api_service_authentication_jwt_failed_attempts_total[5m]))"
+    - expr: sum(rate(api_service_authentication_jwt_failed_attempts_total[5m]))
+      record: aggregation:giantswarm:api_auth_jwt_failed_attempts_total
     # GS Rest API requests
-    - record: aggregation:giantswarm:api_requests
-      expr: "sum(rate(nginx_ingress_controller_requests{namespace='giantswarm', ingress='api'}[5m])) by (status)"
-    - record: aggregation:giantswarm:app_deployed_control_plane_total
-      expr: "count(app_operator_app_info{status=~'deployed|DEPLOYED',namespace='giantswarm'}) by (name,version)"
-    - record: aggregation:giantswarm:app_deployed_tenant_total
-      expr: "count(app_operator_app_info{status=~'deployed|DEPLOYED',namespace!='giantswarm'}) by (name,version)"
-    - record: aggregation:giantswarm:cluster_certificate_not_after_seconds
-      expr: "min(cert_exporter_not_after) by (cluster_id, cluster_type)"
-    - record: aggregation:giantswarm:cluster_release_version
-      expr: "label_replace(sum(cluster_service_cluster_info) by (release_version, cluster_id) / 2, 'cluster_id', '$1', 'cluster_id', '(.*)') or label_replace(sum(cluster_operator_cluster_status{release_version!=''}) by (release_version, cluster_id), 'cluster_id', '$1', 'cluster_id', '(.*)')"
-    - record: aggregation:giantswarm:cluster_transition_create
-      expr: "label_replace(avg_over_time(cluster_operator_cluster_create_transition[1w]), 'cluster_id', '$1', 'cluster_id', '(.*)')"
-    - record: aggregation:giantswarm:cluster_transition_update
-      expr: "label_replace(avg_over_time(cluster_operator_cluster_update_transition[1w]), 'cluster_id', '$1', 'cluster_id', '(.*)')"
+    - expr: sum(rate(nginx_ingress_controller_requests{namespace="giantswarm", ingress="api"}[5m])) by (status)
+      record: aggregation:giantswarm:api_requests
+    - expr: count(app_operator_app_info{status="deployed",namespace="giantswarm"}) by (name,version)
+      record: aggregation:giantswarm:app_deployed_control_plane_total
+    - expr: count(app_operator_app_info{status=~"deployed|DEPLOYED",namespace!="giantswarm"}) by (name,version)
+      record: aggregation:giantswarm:app_deployed_tenant_total
+    - expr: min(cert_exporter_not_after) by (cluster_id, cluster_type)
+      record: aggregation:giantswarm:cluster_certificate_not_after_seconds
+    - expr: label_replace(sum(azure_operator_cluster_release{release_version!=""}) by (release_version, cluster_id), "cluster_id", "$1", "cluster_id", "(.*)") or label_replace(sum(cluster_service_cluster_info) by (release_version, cluster_id) / 2, "cluster_id", "$1", "cluster_id", "(.*)") or label_replace(sum(cluster_operator_cluster_status{release_version!=""}) by (release_version, cluster_id), "cluster_id", "$1", "cluster_id", "(.*)")
```

Comment:
Why did this become so complex ?

Reply:
See https://github.com/giantswarm/g8s-prometheus/pull/1209/files

Reply:
This does not explain the diff seen here

Reply:
Actually, yes as:

`
expr: "label_replace(sum(cluster_service_cluster_info) by (release_version, cluster_id) / 2, 'cluster_id', '$1', 'cluster_id', '(.*)') or label_replace(sum(cluster_operator_cluster_status{release_version!=''}) by (release_version, cluster_id), 'cluster_id', '$1', 'cluster_id', '(.*)')"
`

Became: 

`
- expr: label_replace(sum(azure_operator_cluster_release{release_version!=""}) by (release_version, cluster_id), "cluster_id", "$1", "cluster_id", "(.*)") or label_replace(sum(cluster_service_cluster_info) by (release_version, cluster_id) / 2, "cluster_id", "$1", "cluster_id", "(.*)") or label_replace(sum(cluster_operator_cluster_status{release_version!=""}) by (release_version, cluster_id), "cluster_id", "$1", "cluster_id", "(.*)")
`

in the PR I linked